lib/portage/package/ebuild/doebuild.py

```diff
@@ -2553,6 +2555,37 @@ def _post_src_install_write_metadata(settings):
 
             metadata_buffer[k] = v
 
+    # We could skip this for non-binpkgs but there doesn't seem to be much
+    # value in that, as users shouldn't downgrade libc anyway.
+    #
+    # TODO: We could skip the injection for packages installing no ELF files
+    # by going off the absence of NEEDED, but that runs after this function.
+    # Maybe we should just move the scanelf calls into Python and/or use pyelftools
+    # instead which would let us coalesce the calls too.
+    injected_libc_depstring = []
+    for atom in expand_new_virt(
+        QueryCommand.get_db()[settings["EROOT"]]["vartree"].dbapi,
+        portage.const.LIBC_PACKAGE_ATOM,
+    ):
+        # This will give us something like sys-libs/glibc:2.2, but we want to know
+        # what installed atom actually satifies that.
+        try:
+            libc_realized_atom = QueryCommand.get_db()[settings["EROOT"]][
+                "vartree"
+            ].dbapi.match(atom)[0]
+        except IndexError:
+            continue
+
+        injected_libc_depstring.append(f">={libc_realized_atom}")
+
+    # For RDEPEND, we want an implicit dependency on >=${PROVIDER_OF_LIBC}
+    # to avoid runtime breakage when merging binpkgs, see bug #753500.
+    if injected_libc_depstring and settings.configdict["pkg"].get("RDEPEND"):
```

Comment:
I don't really know what I'm doing here, but I'm continuing testing with
```
if injected_libc_depstring and "RDEPEND" in metadata_buffer:
```

Reply:
And now it's happily adding the dependency in the VDB:
```
pinacolada /var/lib/machines/amd64-stable/var/db/pkg/virtual/perl-Encode-3.190.0 # cat RDEPEND 
|| ( =dev-lang/perl-5.38* ~perl-core/Encode-3.190.0 ) dev-lang/perl:0/5.38= !<perl-core/Encode-3.190.0 !>perl-core/Encode-3.190.0-r999 >=sys-libs/glibc-2.37-r3
```

Reply:
Does it work correctly if the package has no RDEPEND beforehand?

Reply:
Yes, checked that too
```
pinacolada /var/lib/machines/amd64-stable/var/db/pkg/dev-libs/libpipeline-1.5.7 # cat RDEPEND 
>=sys-libs/glibc-2.37-r3
```

Reply:
Hmm, I built a binary package of `app-arch/tar` with this, and now the `Packages` file has the following in the tar entry:
```
RDEPEND: virtual/acl >=sys-libs/glibc-2.37-r3 >=sys-libs/glibc-2.37-r3
```
So somehow the glibc entry duplicated.

Reply:
Confirmed with `zstd`, building it again with binpkg creation doubles the entry.
Does *not* become 3 or 4 entries if I repeat that though.